### ISSUE TITLE
Discord RPC Errors Crash Everest

### ISSUE THREAD

**Issue by WilsontheWolf** [EverestAPI/Everest#699]
### My mods and Everest install are up to date

Yes

### I have recreated the bug with only Everest OR a minimum number of mods enabled

Yes

### Describe the bug

When there is an issues with the RPC Server, Everest will crash. This was caused by me using a non-standard Discord RPC server (https://github.com/WilsontheWolf/activity-tools/tree/master/daemon). Ideally, this should not crash, and instead just behave if there was no RPC to connect to.

### Steps to reproduce

1. Receive an error with the RPC

### Expected behavior

This should either silently continue, as if there was no RPC server.

### Operating System

Fedora 39

### Everest Version

4449

### Additional context

```
Controller 0: Steam Virtual Gamepad, GUID: de28ff11, Mapping: 03000000de280000ff11000001000000,Steam Virtual Gamepad,a:b0,b:b1,back:b6,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,guide:b8,leftshoulder:b4,leftstick:b9,lefttrigger:a2,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b10,righttrigger:a5,rightx:a3,righty:a4,start:b7,x:b2,y:b3,platform:Linux,
CELESTE : 1.4.0.0
(12/09/2023 18:25:13) [Everest] [Info] [core] Booting Everest
(12/09/2023 18:25:13) [Everest] [Info] [core] AppDomain: Celeste.exe+Everest
(12/09/2023 18:25:13) [Everest] [Info] [core] VersionCelesteString: 1.4.0.0-fna [Everest: 4449-azure-ddba3]
(12/09/2023 18:25:13) [Everest] [Info] [core] SystemMemoryMB: 15686.640 MB
(12/09/2023 18:25:13) [Everest] [Info] [discord-game-sdk] Discord Game SDK initialized!
(12/09/2023 18:25:13) [Everest] [Info] [core] Registered code module Celeste.Mod.Core.CoreModule for module Everest 1.4449.0.
(12/09/2023 18:25:14) [Everest] [Info] [core] Non-code module Celeste 1.4.0.0 registered.
(12/09/2023 18:25:14) [Everest] [Info] [core] Non-code module DialogCutscene 1.0.0 registered.
(12/09/2023 18:25:14) [Everest] [Info] [core] Non-code module UpdateChecker 1.0.2 registered.
(12/09/2023 18:25:14) [Everest] [Info] [core] Non-code module InfiniteSaves 1.0.0 registered.
(12/09/2023 18:25:14) [Everest] [Info] [core] Non-code module DebugRebind 1.0.0 registered.
(12/09/2023 18:25:14) [Everest] [Info] [core] Non-code module RebindPeriod 1.0.0 registered.
(12/09/2023 18:25:15) [Everest] [Info] [Everest.LuaBoot] Lua ready.
(12/09/2023 18:25:16) [INF] [LoadSettings] Loaded settings with versioning number '2'.
(12/09/2023 18:25:16) [CelesteNet] [INF] [rc] Started ClientRC thread, available via http://localhost:38038/
(12/09/2023 18:25:16) [Everest] [Info] [core] Registered code module Celeste.Mod.CelesteNet.Client.CelesteNetClientModule for module CelesteNet.Client 2.2.2.
(12/09/2023 18:25:16) [Everest] [Info] [loader] Loading mods with unsatisfied optional dependencies (if any)
Using EXT_swap_control_tear VSync!
FNA3D Driver: OpenGL
OpenGL Renderer: Mesa Intel(R) Graphics (ADL GT2)
OpenGL Driver: 4.6 (Compatibility Profile) Mesa 23.2.1
OpenGL Vendor: Intel
MojoShader Profile: glsl120
BEGIN LOAD
 - GFX LOAD: 1008ms
 - MTN LOAD: 244ms
Using EXT_swap_control_tear VSync!
FULLSCREEN
GAME DISPLAYED (in 1750ms)
--------------------------------
Detailed exception log:
--------------------------------
Discord.ResultException: InternalError
  at Celeste.Mod.Everest+DiscordSDK.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x00101] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Microsoft.Xna.Framework.Game.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x0006a] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Monocle.Engine.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x001dd] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Celeste.Celeste.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x00036] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Microsoft.Xna.Framework.Game.Tick () [0x00102] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Microsoft.Xna.Framework.Game.RunLoop () [0x00019] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Microsoft.Xna.Framework.Game.Run () [0x00032] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Monocle.Engine.RunWithLogging () [0x00000] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
 - AUDIO LOAD: 536ms
 - GFX DATA LOAD: 118ms
 - MTN DATA LOAD: 79ms
 - DIA/FONT LOAD: 404ms
 - MODDED MTN LOAD: 3ms
 - MODDED MTN DATA LOAD: 1ms
 - LEVELS LOAD: 1179ms
(12/09/2023 18:25:33) [Everest] [Info] [discord-game-sdk] Discord Game SDK disposed
```

```
Celeste Error Log
==========================================

Ver 1.4.0.0-fna [Everest: 4449-azure-ddba3]
12/09/2023 18:25:18
Discord.ResultException: InternalError
  at Celeste.Mod.Everest+DiscordSDK.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x00101] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Microsoft.Xna.Framework.Game.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x0006a] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Monocle.Engine.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x001dd] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Celeste.Celeste.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x00036] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Microsoft.Xna.Framework.Game.Tick () [0x00102] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Microsoft.Xna.Framework.Game.RunLoop () [0x00019] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Microsoft.Xna.Framework.Game.Run () [0x00032] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Monocle.Engine.RunWithLogging () [0x00000] in <4a26f9ded6704c87a2f47e66d2d85163>:0 


Ver 1.4.0.0-fna [Everest: 4449-azure-ddba3]
12/09/2023 00:30:47
System.NullReferenceException: Object reference not set to an instance of an object
  at Monocle.TagLists.EntityAdded (Monocle.Entity entity) [0x00004] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Monocle.EntityList.UpdateLists () [0x0004f] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Celeste.Level.Update () [0x0034e] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Monocle.Engine.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x00145] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Celeste.Celeste.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x00036] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Microsoft.Xna.Framework.Game.Tick () [0x00102] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Microsoft.Xna.Framework.Game.RunLoop () [0x00019] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Microsoft.Xna.Framework.Game.Run () [0x00032] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Monocle.Engine.RunWithLogging () [0x00000] in <4a26f9ded6704c87a2f47e66d2d85163>:0 


Ver 1.4.0.0-fna [Everest: 4449-azure-ddba3]
12/08/2023 01:37:54
Discord.ResultException: InternalError
  at Celeste.Mod.Everest+DiscordSDK.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x00101] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Microsoft.Xna.Framework.Game.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x0006a] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Monocle.Engine.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x001dd] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Celeste.Celeste.Update (Microsoft.Xna.Framework.GameTime gameTime) [0x00036] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
  at Microsoft.Xna.Framework.Game.Tick () [0x00102] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Microsoft.Xna.Framework.Game.RunLoop () [0x00019] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Microsoft.Xna.Framework.Game.Run () [0x00032] in <6eeb4fc8c61b441fb2d2fd16aa9405a2>:0 
  at Monocle.Engine.RunWithLogging () [0x00000] in <4a26f9ded6704c87a2f47e66d2d85163>:0 
```

**Comment by ErynGalen** [comment]
Should be fixed by #768
Testing would be appreciated.

**Comment by Kalobi** [comment]
Fixed by 4155fc57a9511a7124a6a28171a5c66d7e35525b
